### ISSUE TITLE
.editorconfig "Value cannot be null. Parameter name: path"  at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.<>c.<G

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#58732]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/editorconfig-Value-cannot-be-null-Par/1588158)._

---
[severity:It's more difficult to complete my work] [regression] [worked-in:VS2019]
Our project is set up like this:

```text
C:\project\.editorconfig
C:\project\example\.editorconfig
```

When I try to open `C:\project\example\.editorconfig`, VS pops up an error message:
```text
Value cannot be null.
Parameter name: path
```

If I then quickly try to open the file again, VS pops up a different dialog with a more detailed error message:
```text
An error occurred while initializing the frame's content
System.ArgumentNullException: Value cannot be null.
Parameter name: path
   at System.IO.DirectoryInfo.. ctor(String path)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.<>c.<GetProjectsForPath>b__0_1(Project p)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.GetProjectsForPath(Solution solution, String givenPath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.SettingsProviderBase`4.Update()
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.Whitespace.CommonWhitespaceSettingsProviderFactory.GetForFile(String filePath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.CombinedOptionsProviderFactory`1.GetForFile(String filePath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.SettingsAggregator.GetSettingsProvider[TData](String fileName)
   at Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.SettingsEditorPane.<>c__DisplayClass12_0. <Initialize>g__GetWhitespaceView|1()
   at Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.SettingsEditorPane.Initialize()
   at Microsoft.VisualStudio.Shell.WindowPane.InternalSetSite(IServiceProvider p)
   at Microsoft.VisualStudio.Platform.WindowManagement.UIElementDocumentObject.SetSite(DocumentObjectSite site)
   at Microsoft.VisualStudio.Platform.WindowManagement.DocumentObjectSite.InitializeDocumentObject(Object punkView)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.InitializeDocumentSite(Boolean creatingStubFrame, Boolean replacingStubView, Object punkView, Object punkData, IServiceProvider pServiceProvider, IVsUIHierarchy pUIHierarchy, UInt32 vsid)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowManagerService.CreateContentPane(FrameMoniker frameMoniker, Boolean isDocument, String lpstrMkDoc, UInt32 eCreateWindowFlags, Object punkView, Object punkData, IServiceProvider pServiceProvider, IVsUIHierarchy pUIHierarchy, UInt32 vsid, Guid rguidCmdUI, ViewGroup parent, IVsWindowFrame& ppWindowFrame)

File path: C:\project\example\.editorconfig
Frame GUID: ...
Frame mode: VSFM_Float
Error code: 0x80004003
```

See attached recording for a video showing the issue.

Let me know if there's any more info that I can provide! 

---
#### Feedback Bot on 11/18/2021, 00:48 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Feedback Bot on 11/18/2021, 10:14 AM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

#### Jonathon Marolf [MSFT] on 11/18/2021, 10:48 AM: 

<p>A fix for this issue has been internally implemented and is being prepared for release (you can find the pull request <a target='_blank' href="https://github.com/dotnet/roslyn/pull/57234">here</a>). We’ll update you once it becomes available for download.</p>

---
### Original Solutions
(no solutions)

**Comment by Youssef1313** [comment]
cc @jmarolf

**Comment by jmarolf** [comment]
confirmed this is fixed with https://github.com/dotnet/roslyn/pull/57234. Issue is the fix is in 17.1 which has not been released yet.
